### DISCUSSION TITLE
How to Perform Measurements Using HydraHarp Control Panel Settings via API

### DISCUSSION THREAD

**Discussion by Brodis82** [PicoQuant/snAPI#29] [Q&A] [answered]
Hello,

I am currently working with the HydraHarp Control Panel and need to replicate the settings for a measurement.
I am using HydraHarp 400.
 I have attached images of the control panel with the specific parameters I want to use.

The settings are as follows:

HydraHarp Axis Panel:

Time Axis:
Min: 0 ns
Max: 100 ns
Count Axis:
Min: 100
Max: 1000000
Mode: Log
HydraHarp Control Panel (Sync Input):

ZeroCr: 10 mV
Divider: None
Discr.: 50 mV
Offset: -1000 ps
HydraHarp Control Panel (Inputs 1.4):

Input Channels 1-4:
ZeroCr: 10 mV
Discr.: 50 mV
Offset: 0 ps
HydraHarp Control Panel (Acquisition):

Resolution: 64 ps
Time: 5 s
Offset: 0 ns
Trce/Block: 0
Mode: INT
Stop at: 4294967295 cnt
Could someone guide me on how to configure these parameters and perform a measurement using the API?

Thank you!

![image axis](https://github.com/PicoQuant/snAPI/assets/94947586/51bc8cbe-6631-46a2-9a63-c8d79e487fad)
![sync](https://github.com/PicoQuant/snAPI/assets/94947586/8f2ae589-db20-4635-8642-1fd6870a0c75)
![image 1-4](https://github.com/PicoQuant/snAPI/assets/94947586/1adf3255-959f-4343-99c1-19e8e7b80647)
![acquistion](https://github.com/PicoQuant/snAPI/assets/94947586/95e7f1ae-7bfc-486a-9395-6349a4e33cfd)

**Comment by tpoint75** [comment, marked as answer]
Time Axis:
Min: 0 ns
Max: 100 ns
Count Axis:
Min: 100
Max: 1000000
Mode: Log
>The axis must be set in the plot (plt.xxx this is part of the matplotlib - read the documentation)

HydraHarp Control Panel (Sync Input):
ZeroCr: 10 mV
Divider: None
Discr.: 50 mV
Offset: -1000 ps

HydraHarp Control Panel (Inputs 1.4):
Input Channels 1-4:
ZeroCr: 10 mV
Discr.: 50 mV
Offset: 0 ps

>can be set in the config.ini:
[config](https://picoquant.github.io/snAPI/config.html)

>The rest is part of the python script. I think this would be a good start:
[Demo_HistogramSimple](https://github.com/PicoQuant/snAPI/blob/main/demos/Demo_HistogramSimple.py)
